recipes/libhal/all/conanfile.py

```diff
@@ -34,9 +32,19 @@ def _compilers_minimum_version(self):
             "Visual Studio": "17",
             "msvc": "193",
             "clang": "13",
-            "apple-clang": "13.1.6"
+            "apple-clang": "14.0.0"
         }
 
+    def layout(self):
+        basic_layout(self, src_folder="src")
+
+    def requirements(self):
+        if Version(self.version) >= "2.0.0":
+            self.requires("boost-leaf/1.82.0", transitive_headers=True)
```

Comment:
I believe the `boost-leaf` package is deprecated as of #16536.

```suggestion
            self.requires("boost/1.83.0", transitive_headers=True)
```